src/spac/transformations.py

```diff
@@ -280,47 +279,50 @@ def batch_normalize(adata, annotation, layer, method="median", log=False):
 
     """
     allowed_methods = ["median", "Q50", "Q75"]
-    regions = adata.obs[annotation].unique().tolist()
+    batches = adata.obs[annotation].unique().tolist()
     original = adata.to_df()
 
     if log:
         original = np.log2(1+original)
 
     if method == "median" or method == "Q50":
-        all_regions_quantile = original.quantile(q=0.5)
+        all_batch_quantile = original.quantile(q=0.5)
+        print("Median for all cells:", all_batch_quantile)
     elif method == "Q75":
-        all_regions_quantile = original.quantile(q=0.75)
+        all_batch_quantile = original.quantile(q=0.75)
+        print("Q75 for all cells:", all_batch_quantile)
     else:
         raise Exception(
             "Unsupported normalization {0}, allowed methods = {1]",
             method, allowed_methods)
 
-    # Place holder for normalized dataframes per region
-    new_df_list = []
-    for region in regions:
-        region_cells = original[adata.obs[annotation] == region]
+    # Place holder for normalized dataframes per batch
+    for batch in batches:
+        batch_cells = original[adata.obs[annotation] == batch]
+        print(f"Processing batch: {batch}, original values:\n", batch_cells)

```

Comment:
@fangliu117 , would you please use the "logging" package instead of "print"?

Reply:
Thank you @georgezakinih I will fix it now.